src/sentry/shared_integrations/client/base.py

```diff
@@ -329,3 +371,104 @@ def get_with_pagination(
             if num_results < page_size:
                 return output
         return output
+
+    def record_response(self, response: BaseApiResponse):
+        if self.is_response_fatal(response):
+            self.record_request_fatal(response)
+        elif self.is_response_error(response):
+            self.record_request_error(response)
+        elif self.is_response_success(response):
+            self.record_request_success(response)
+
+    def record_error(self, error: Exception):
+        redis_key = self._get_redis_key()
+        if not len(redis_key):
+            return
+        if not self.is_considered_error(error):
+            return
+        buffer = IntegrationRequestBuffer(redis_key)
+        buffer.record_error()
```

Comment:
You can probably put your try .. except here - anywhere you construct an `IntegrationRequestBuffer` and call its methods. Maybe something like:

```
from sentry.utils import metrics

try:
  buffer = IntegrationRequestBuffer(redis_key)
  buffer.record_error()
except Exception:
  metrics.incr("integration.slack.disable_on_broken.redis")
  return
```